src/Service.Tests/SqlTests/GraphQLMutationTests/DwSqlGraphQLMutationTests.cs

```diff
@@ -434,6 +434,154 @@ public override async Task DeleteMutationWithOnlyTypename()
             JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
             SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
         }
+
+        /// <summary>
+        /// <code>Do: </code>insert a new book and return all the books with same publisher
+        /// <code>Check: </code>the intended book is inserted.
+        /// </summary>
+        [TestMethod]
+        public async Task TestStoredProcedureMutationNonEmptyResponse()
+        {
+            string dbQuery = @"
+SELECT CASE
+         WHEN EXISTS (
+           SELECT 1
+           FROM [books] AS [table0]
+           JOIN (
+             SELECT id
+             FROM [publishers]
+             WHERE name = 'Big Company'
+           ) AS [table1] ON [table0].publisher_id = [table1].id
+         ) THEN 'true' ELSE 'false'
+       END AS IsInserted
+            ";
+
+            string graphQLMutationName = "executeInsertAndDisplayAllBooksUnderGivenPublisher";
+            string graphQLMutation = @"
+                mutation{
+                    executeInsertAndDisplayAllBooksUnderGivenPublisher(book_id: 1, title: ""Orange Tomato"" publisher_name: ""Big Company""){
+                        id
+                        title
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+            string insertedValue = "{\"id\":1,\"title\":\"Orange Tomato\"}";
+            string expected = await GetDatabaseResultAsync(dbQuery);
+            Assert.IsTrue(actual.ToString().Contains(insertedValue));
+            Assert.IsTrue(expected.ToString().Equals("True"));
+        }
+
+        /// <summary>
+        /// <code>Do: </code>insert new Book and return nothing
```

Comment:
return nothing? the selection set specifies the field `result`

Reply:
Updated summary.